### PR TITLE
ARROW-13216: [R] Type checks test fails with rtools35

### PR DESCRIPTION
**Pull request by ianmcook**

### CODE REVIEW

**Comment by ianmcook** [comment]
@github-actions crossbow submit test-r-rtools-35

File: r/tests/testthat/test-dplyr.R [hunk 1]
```diff
@@ -815,7 +815,18 @@ test_that("type checks on expressions", {
         a = is.character(as.character(int)),
         b = is.integer(as.character(int)),
         c = is.integer(int + int),
-        d = is.double(int + dbl),
+        d = is.double(int + dbl)
```
Comment (jonkeane) [inline r/tests/testthat/test-dplyr.R:818]: If you wanted to confirm that `is.logical` works you could do:

```
e = is.logical(chr %in% c("d", "e", "f"))
```

**Review by jonkeane — APPROVED** [review]
This looks good, one small comment if we wanted to simplify the test a bit. Though I like that we're testing `grepl` since that's common and goes through a slightly different NSE route than the others IIRC

File: r/tests/testthat/test-dplyr.R (hunk 1 above)
Comment (ianmcook) [reply r/tests/testthat/test-dplyr.R:818]: Sounds good. I'll keep it simpler and use a numeric inequality instead.

**Comment by ianmcook** [comment]
@github-actions crossbow submit test-r-rtools-35

**Pull request closed without merging**